src/videocontext.ts

```diff
@@ -1062,10 +1076,32 @@ export default class VideoContext {
                 }
             }
 
+            const ready = !this._isStalled();
+            const isDirty = sortedNodes.some((node) => node.isDirty);
+            const timeChanged = this._prevTime !== this._currentTime;
+            const renderDirtyNodes =
+                ready && (isDirty || timeChanged) && this._renderOnDirtyNodeOnly;
+
             for (let node of sortedNodes) {
-                if (this._sourceNodes.indexOf(node as SourceNode) === -1) {
-                    (node as ProcessingNode)._update(this._currentTime);
-                    (node as ProcessingNode)._render();
+                if (renderDirtyNodes) {
+                    // make sure nodes are updated before rendering the processing nodes
+                    if (node instanceof SourceNode || node instanceof ProcessingNode) {
+                        node._update(this._currentTime);
+                    }
+                }
+                if ((this._sourceNodes as GraphNode[]).indexOf(node) === -1) {
+                    if (
+                        !this._renderOnDirtyNodeOnly ||
+                        this._state === VideoContext.STATE.PLAYING
+                    ) {
+                        (node as ProcessingNode)._update(this._currentTime);
+                        (node as ProcessingNode)._render();
+                    } else if (renderDirtyNodes) {
+                        (node as ProcessingNode)._update(this._currentTime);
+                        (node as ProcessingNode)._render();
+                        node.isDirty = false;
```

Comment:
I think this will only mark processing nodes the node as no longer dirty. We should add a 
```
if (renderDirtyNodes) {
    node.isDirty = false;
}
```

at the end of the loop.